src/NuGet.Clients/NuGet.PackageManagement.UI/Models/NuGetProjectUpgradeWindowModel.cs

```diff
@@ -0,0 +1,247 @@
+            bool collapseDependencies)
+        {
+            PackageDependencyInfos = packageDependencyInfos;
+            Project = project;
+            _collapseDependencies = collapseDependencies;
+        }
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        public NuGetProject Project { get; }
+
+        public string Title => string.Format(CultureInfo.CurrentCulture, Resources.Text_ChangesForProject, ProjectName);
+
+        private string ProjectName
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_projectName))
+                {
+                    _projectName = NuGetProject.GetUniqueNameOrName(Project);
+                    return _projectName;
+                }
+                else
+                {
+                    return _projectName;
+                }
+            }
+            set
+            {
+                _projectName = value;
+            }
+        }
+
+        public IList<PackageDependencyInfo> PackageDependencyInfos { get; }
+
+        // Changing CollapseDependencies updates the list of included packages
+        public bool CollapseDependencies
+        {
+            get { return _collapseDependencies; }
+            set
+            {
+                _collapseDependencies = value;
+                OnPropertyChanged(nameof(DirectDependencies));
+                OnPropertyChanged(nameof(TransitiveDependencies));
+            }
+        }
+
+        public IEnumerable<string> Warnings
+        {
+            get
+            {
+                if (_warnings == null)
+                {
+                    InitPackageUpgradeIssues();
+                }
+                return _warnings;
+            }
+        }
+
+        public IEnumerable<string> Errors
+        {
+            get
+            {
+                if (_errors == null)
+                {
+                    InitPackageUpgradeIssues();
+                }
+                return _errors;
+            }
+        }
+
+        public IList<PackageIdentity> NotFoundPackages
+        {
+            get
+            {
+                if (_notFoundPackages == null)
+                {
+                    InitPackageUpgradeIssues();
+                }
+                return _notFoundPackages;
+            }
+        }
+
+        public bool HasIssues => Errors.Any() || Warnings.Any();
+
+        public IEnumerable<NuGetProjectUpgradeDependencyItem> UpgradeDependencyItems
+            => _upgradeDependencyItems ?? (_upgradeDependencyItems = GetUpgradeDependencyItems());
+
+        public IEnumerable<string> DirectDependencies => CollapseDependencies ? IncludedCollapsedPackages : AllPackages;
+
+        public IEnumerable<string> TransitiveDependencies => CollapseDependencies ? DependencyPackages : new List<string>();
+
+        private IEnumerable<string> DependencyPackages => _dependencyPackages ?? (_dependencyPackages = GetDependencyPackages());
```

Comment:
`DependencyPackages`, `AllPackages`, and `IncludedCollapsedPackages` should also be lazy initialized.